cf-agent/verify_files_utils.c

```diff
@@ -1577,8 +1578,19 @@ bool CopyRegularFile(EvalContext *ctx, const char *source, const char *dest, con
         }
     }
     else
-    {
 #endif
+    {
+        // Retain destination file's permissions
+        if (!attr->copy.preserve)
+        {
+            int mode = backupok ? (s.st_mode & 0777) : CF_PERMS_DEFAULT;
```

Comment:
What does the mode have to do with `backupok`? And what is this `& 0777` supposed to limit? SUID bit?

Reply:
Well, it's not well explained. `backupok` means that there was a file that will be overwritten (but there was a backup). If there was no file, the mode is set to `CF_PERMS_DEFAULT`.

I think I might work a bit more on this PR and create a new or modify the existing copy function that also takes permissions so I don't change the mode _after_ the file is created. Just to get rid of a bit of technical debt.

Reply:
> And what is this `& 0777` supposed to limit? SUID bit?

And yeah, was thinking the SUID bit.